packages/@o3r/schematics/src/utility/metrics.ts

```diff
@@ -0,0 +1,37 @@
+import { callRule, Rule } from '@angular-devkit/schematics';
+import { performance } from 'node:perf_hooks';
+import { lastValueFrom } from 'rxjs';
+import { getEnvironmentInfo } from './environment';
+
+export const createSchematicsWithMetrics = <S>(
+  schematicsFn: (opts: S) => Rule
+): ((opts: S) => Rule) => (options: S) => async (tree, context) => {
+    const startTime = Math.floor(performance.now());
+    let error;
+    try {
+      await lastValueFrom(callRule(schematicsFn(options), tree, context));
+    }
+    catch (e: any) {
+      error = e.toString();
+      throw (e instanceof Error ? e : new Error(error));
```

Comment:
do we want to log metrics in case of errors ? 
in that case you should move the throw at the end of the function

Reply:
as discussed, the `finally` block will be executed even with the throw, ignore my comment :)